core/src/indexer/fetch_logs.rs

```diff
@@ -36,99 +39,249 @@ pub fn fetch_logs_stream(
-
-        // Live indexing mode
-        if config.live_indexing && !force_no_live_indexing {
-            live_indexing_stream(
-                &config.network_contract.cached_provider,
-                &tx,
-                &contract_address,
-                &config.topic_id,
-                &config.indexing_distance_from_head,
-                current_filter,
+            info!(
+                "{} - {} - Finished indexing historic events",
                 &config.info_log_name,
-                &config.semaphore,
-                config.network_contract.disable_logs_bloom_checks,
-            )
-            .await;
+                IndexingEventProgressStatus::Completed.log()
+            );
+
+            // Live indexing mode
+            if config.live_indexing && !force_no_live_indexing {
+                live_indexing_stream(
+                    &config.network_contract.cached_provider,
+                    &tx,
+                    &contract_address,
+                    &config.topic_id,
+                    &config.indexing_distance_from_head,
+                    current_filter,
+                    &config.info_log_name,
+                    &config.semaphore,
+                    config.network_contract.disable_logs_bloom_checks,
+                )
+                .await;
+            }
         }
     });
 
     UnboundedReceiverStream::new(rx)
 }
 
+async fn process_exex_stream(
+    reth_tx: Arc<ExExTx>,
+    tx: &mpsc::UnboundedSender<Result<FetchLogsResult, Box<dyn Error + Send>>>,
+    config: &EventProcessingConfig,
+) {
+    let reth_tx = reth_tx.clone();
+    // TODO: Since most of the codebase uses ethers types, and we want reth types here,
+    // we are converting ethers types to reth types here.
+    // we should move towards using reth types throughout the codebase.
+    let filter = config.to_event_filter().unwrap().to_alloy_filter();
+
+    // Process backfill data
+    let to_block = config.to_event_filter().unwrap().get_to_block().as_u64();
+
+    let (response_tx, response_rx) = oneshot::channel();
+
+    let res = reth_tx.send(ExExRequest::Start {
+        mode: ExExMode::HistoricOnly,
+        filter: filter.clone(),
+        response_tx,
+    });
+    if let Err(e) = res {
+        error!(
+            "{} - {} - Failed to start backfill: {}",
+            config.info_log_name,
+            IndexingEventProgressStatus::Syncing.log(),
+            e
+        );
+        return;
+    };
+
+    let (job_id, rx) = if let Ok(Ok(res)) = response_rx.await {
+        res
+    } else {
+        error!(
+            "{} - {} - Failed to start backfill",
+            config.info_log_name,
+            IndexingEventProgressStatus::Syncing.log(),
+        );
+        return;
+    };
+    info!("Backfill started for job {}", job_id);
+    let mut batched_stream = UnboundedReceiverStream::new(rx).chunks(100);
+    while let Some(logs) = batched_stream.next().await {
+        let mut from_block: u64 = u64::MAX;
```

Comment:
is there ever a place where you understand that you got all the backfill to say block x aka like how we create the internal rindexer db tables to check last synced block number to avoid going through them all on every startup?

Reply:
last synced block still is being written because we are still returning wrapped logs back to the stream. So the `fetch_logs_stream` would restart with the last synced block from the config, as per existing functionality. So here the stream would restart with the last synced block itself